server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java

```diff
@@ -473,6 +482,79 @@ public void testScheduledFixedDelayForceExecution() {
         }
     }
 
+    public void testThreadPoolMetrics() {
+        final RecordingMeterRegistry meterRegistry = new RecordingMeterRegistry();
+        final BuiltInExecutorBuilders builtInExecutorBuilders = new DefaultBuiltInExecutorBuilders();
+
+        final ThreadPool threadPool = new ThreadPool(
+            Settings.builder().put(Node.NODE_NAME_SETTING.getKey(), "test").build(),
+            meterRegistry,
+            builtInExecutorBuilders
+        );
+        try {
+            final ThreadPool.Info genericInfo = threadPool.info(ThreadPool.Names.GENERIC);
+
+            meterRegistry.getRecorder().collect();
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_ACTIVE, 0);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_CURRENT, 0);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_LARGEST, 0);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_QUEUE, 0);
+            assertGenericLongAsyncCounterValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_COMPLETED, 0);
+            assertGenericDoubleGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_UTILISATION, 0.0d);
+
+            final CyclicBarrier barrier = new CyclicBarrier(2);
+            Future<?> future = threadPool.generic().submit(() -> {
+                safeAwait(barrier);
+                safeAwait(barrier);
+            });
+            safeAwait(barrier);
+
+            meterRegistry.getRecorder().collect();
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_ACTIVE, 1);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_CURRENT, 1);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_LARGEST, 1);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_QUEUE, 0);
+            assertGenericLongAsyncCounterValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_COMPLETED, 0);
+            assertGenericDoubleGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_UTILISATION, (double) 1 / genericInfo.getMax());
+
+            // Let the task complete
+            safeAwait(barrier);
+            safeGet(future);
+
+            meterRegistry.getRecorder().collect();
+            assertGenericLongAsyncCounterValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_COMPLETED, 1);
+            assertGenericLongGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_ACTIVE, 0);
+            assertGenericDoubleGaugeValue(meterRegistry, ThreadPool.THREAD_POOL_METRIC_NAME_UTILISATION, 0.0);
+        } finally {
+            ThreadPool.terminate(threadPool, 10, TimeUnit.SECONDS);
+        }
+    }
```

Comment:
This could be more comprehensive, but the aim here was really just to test the new utilisation metric